### PR TITLE
[ELK] Add GitHub stats support

### PR DESCRIPTION
**Pull request by valeriocos**
This PR provides support to track GitHub repository stats such as the evolution of forks, stars and watchers. Test have been included.

Find below an example about how to use it from mordred:
**setup.cfg**
```
[githubstats]
raw_index = test_github-repo-raw
enriched_index = test_github-repo
api-token = ....
sleep-for-rate = true
no-archive = true
category = repository
```

**projects.json**
```
"githubstats": [
   "https://github.com/chaoss/grimoirelab-perceval"
]
```

### CODE REVIEW

File: grimoire_elk/enriched/github_stats.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+#
```
Comment (sduenas) [inline grimoire_elk/enriched/github_stats.py]: Remove this blank line and update the copyright year to include 2019 too.

Comment (sduenas) [reply grimoire_elk/enriched/github_stats.py]: Why is this in a different module? With GitLab categories are managed on the same module.

File: grimoire_elk/raw/github_stats.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+#
+# StackExchange Ocean feeder
```
Comment (sduenas) [inline grimoire_elk/raw/github_stats.py]: Same comments as in the enrichment module. Moreover, remove this comment. This is not related to StackExchange.

File: tests/test_githubstats.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        eitem = enrich_backend.get_rich_item(item)
+        self.assertEqual(eitem['forks'], 1)
+        self.assertEqual(eitem['stars'], 1)
+        self.assertEqual(eitem['watchers'], 1)
+
+    def test_raw_to_enrich_projects(self):
+        """Test enrich with Projects"""
+
+        result = self._test_raw_to_enrich(projects=True)
+        # ... ?
```
Comment (sduenas) [inline tests/test_githubstats.py]: Why is this comment here and on line 92?

**Review by sduenas — CHANGES_REQUESTED** [review]
Please address my comments.

File: grimoire_elk/enriched/github_stats.py (hunk 1 above)
Comment (valeriocos) [reply grimoire_elk/enriched/github_stats.py]: GitLab merges and issues share similar mappings (the one that we redefined is common in both), similar thing happens for GitHub issues and pull requests.  Based on the previous implementation of the PR in perceval (https://github.com/chaoss/grimoirelab-perceval/pull/481), the GitHub repo data didn't share things with the issue and pull request mappings, furthermore it didn't include identities information. However, with the current PR in perceval (the repo data includes identities such as `owner`) and if we are fine of having mappings about attributes which don't exist in the data (e.g., `merge_author_geolocation`, `assignee_geolocation`, etc.), this PR can be reworked to add everything in the same module.

Comment (valeriocos) [reply grimoire_elk/enriched/github_stats.py]: > Remove this blank line and update the copyright year to include 2019 too.
If you agree, I would address this comment in another PR, since it seems this applies to most files in ELK.

File: grimoire_elk/raw/github_stats.py (hunk 2 above)
Comment (valeriocos) [reply grimoire_elk/raw/github_stats.py]: Perfect, if you agree I would address this comment in another PR (the same as above), since Ocean is an old definition for raw connectors.

File: tests/test_githubstats.py (hunk 3 above)
Comment (valeriocos) [reply tests/test_githubstats.py]: When tests were introduced in ELK, these ones were kickstarted with the idea of completing them at some point in the future (same situation in the other tests), but the task was never scheduled.
